### PR TITLE
[SYCL][Graph] Update spec supported features

### PR DESCRIPTION
**Pull request by EwanC**
The following features are defined in the specification as unsupported, but have working implementations upstream.

* https://github.com/intel/llvm/pull/11418
* https://github.com/intel/llvm/pull/11505
* https://github.com/intel/llvm/pull/11556 (should be merged soon)

### CODE REVIEW

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc [hunk 1]
```diff
@@ -1164,11 +1164,11 @@ passed an invalid event.
...
-cannot be used in graph nodes. A synchronous exception will be thrown with
-error code `invalid` if a user tries to add them to a graph.
-
-Removing this restriction is something we may look at for future revisions of
-`sycl_ext_oneapi_graph`.
+can only be used in graph nodes created using the Record & Replay API, as
+barriers rely on events to enforce dependencies. A synchronous exception will be
+thrown with error code `invalid` if a user tries to add them to a graph using
+the Explicit API. Empty nodes can be used instead of barriers when a user is
+building a graph with the explicit API.
```
Comment (mfrancepillois) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc]: Currently if users using the Explicit API want to implement a barrier that waits for all previous nodes to complete, they have to create an empty node that explicitly depends on all previous nodes. Does it make sense to change the definition of empty nodes such that if users add an empty node with no dependencies, this node will automatically take all previous nodes as dependencies (general barrier)? Or, if we want to keep a empty node without dependency (not sure why should be necessary?) to propose a wildcard (shortcut) to add all previous nodes as dependencies?

Comment (EwanC) [reply sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc]: That idea makes sense to me 👍  keeping the existing empty node with no dependency semantics doesn't really add any value to the user, but making it depend on previous leaf nodes corresponds to barrier semantics to is useful and better aligns with the message of an alternative to the barrier extension.

Comment (Bensuo) [reply sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc]: I feel like having that automatic creation of dependencies on the explicit API is a little bit counter to the existing behaviour where the explicit API doesn't do things like this in the background. This perhaps makes it a bit unintuitive.

Comment (julianmi) [reply sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc]: I tend to agree with @Bensuo here. However, we might want to add a barrier shortcut if this is frequently used in applications.

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc [hunk 2]
```diff
@@ -1737,10 +1737,8 @@ if used in application code.
 . Using `handler::memset` in a graph node.
 . Using `handler::prefetch` in a graph node.
 . Using `handler::memadvise` in a graph node.
-. Using specialization constants in a graph node.
 . Using reductions in a graph node.
 . Using sycl streams in a graph node.
-. Using a kernel bundle in a graph node.
 . Profiling an event returned from graph submission with
   `event::get_profiling_info()`.
```
Comment (EwanC) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc]: I should also add a bullet point here about passing the no immediate command-list property on queue creation to workaround current issue with immediate command-lists.

**Pull request closed without merging**